packages/mattermost-redux/src/selectors/entities/search.ts

```diff
@@ -26,3 +27,28 @@ export const getAllUserMentionKeys: (state: GlobalState) => UserMentionKey[] = c
         return userMentionKeys.concat(groupMentionKeys);
     },
 );
+
+export function recentSearches(state: GlobalState): SearchParams[] {
+    const recentSearches = state.entities.search.recentSearches;
+    const channels = state.entities.channels.channels;
+    const userProfiles = state.entities.users.profiles;
+    if (!recentSearches) {
+        return [];
+    }
+    return recentSearches.map((searchParams) =>
+        ({
+            ...searchParams,
+            in_channels: searchParams?.in_channels?.
+                map((channelId) => channels[channelId as unknown as string] || null).
```

Comment:
I don't think the `|| null` should be necessary here, especially if you replace the `filter` with `filter(Boolean)`